### PR TITLE
[Bug] fix training session encounter not applying nature correctly

### PR DESCRIPTION
**Pull request by dp0ch**
## What are the changes the user will see?
Fixed a bug where the training session encounter would not apply the nature for the current run if the pokemon had a nature set on `customPokemonData` e.g. via mints or the vitamin dealer encounter

## Why am I making these changes?
I found this bug and reported it here https://discord.com/channels/1125469663833370665/1303191649723093032

## What are the changes from a developer perspective?
Set `customPokemonData.nature` in the training session encounter

## How to test the changes?

- Set overrides 
```js
{
  MYSTERY_ENCOUNTER_OVERRIDE: MysteryEncounterType.TRAINING_SESSION,
  MYSTERY_ENCOUNTER_RATE_OVERRIDE: 256,
  STARTING_LEVEL_OVERRIDE: 20,
  ITEM_REWARD_OVERRIDE: [{ name: "MINT" }],
  STARTING_WAVE_OVERRIDE: 9,
}
```
- Apply mint
- Change nature via training session encounter

### CODE REVIEW

File: src/data/mystery-encounters/encounters/training-session-encounter.ts [hunk 1]
```diff
@@ -226,6 +226,7 @@ export const TrainingSessionEncounter: MysteryEncounter =
             queueEncounterMessage(scene, `${namespace}:option.2.finished`);
             // Add the pokemon back to party with Nature change
             playerPokemon.setNature(encounter.misc.chosenNature);
+            playerPokemon.customPokemonData.nature = encounter.misc.chosenNature;
```
[the remark this answers is not in the document]
Comment (dp0ch) [reply src/data/mystery-encounters/encounters/training-session-encounter.ts]: I implemented your suggestion. The encounter doesn't set the base nature anymore. Mints and the training session ME should now correctly update stats. I also added a utility method to unlock a nature directly without calling `gameData.setPokemonSpeciesCaught` as this currently unlocks the base nature.

**Comment by MokaStitcher** [comment]
Finally getting back to this, sorry. Would you mind updating the PR description with the new changes please?

File: src/system/game-data.ts [hunk 2]
```diff
@@ -1791,6 +1791,35 @@ export class GameData {
     });
   }
 
+  /**
+   * @param {PokemonSpecies} species
+   * @returns {boolean} whether root species has been unlocked for given {@linkcode PokemonSpecies}
```
Comment (MokaStitcher) [inline src/system/game-data.ts:1796]:
```suggestion
   * Checks whether the root species of a given {@PokemonSpecies} has been unlocked in the dex
```

File: src/system/game-data.ts [hunk 3]
```diff
@@ -1791,6 +1791,35 @@ export class GameData {
...
+   */
+  isRootSpeciesUnlocked(species: PokemonSpecies): boolean {
+    return !!this.dexData[species.getRootSpeciesId()]?.caughtAttr;
+  }
+
+  /**
+   * Unlocks nature for a given {@linkcode PokemonSpecies} and its prevolutions.
+   * Will fail silently if root species has not been unlocked
+   * @param {PokemonSpecies} species
+   * @param {Nature} nature
```
Comment (MokaStitcher) [inline src/system/game-data.ts:1806]:
```suggestion
   * Unlocks the given {@linkcode Nature] for a {@linkcode PokemonSpecies} and its prevolutions.
   * Will fail silently if root species has not been unlocked
```
No need to add the @ params if they are self explanatory

**Review by MokaStitcher — COMMENTED** [review]
lgtm. Just a couple suggestions for consistency with our usual comment style
(tested with Pikachu, mint still works as before and unlocks nature for Pikachu and Pichu. Mint-given nature gets overrided by training session, all seems good)

**Review by MokaStitcher — APPROVED** [review]

**Pull request merged**